viz_scripts/survey_responses.ipynb

```diff
@@ -0,0 +1,343 @@
+    "all_composite_trips = scaffolding.filter_composite_trips(all_composite_trips, program, include_test_users)\n",
+    "\n",
+    "if len(all_composite_trips) > 0:\n",
+    "    #remove blank inputs\n",
+    "    survey_trips = all_composite_trips[all_composite_trips['user_input'] != {}]\n",
+    "\n",
+    "    #survey counts df\n",
+    "    survey_trips = survey_trips.reset_index()\n",
+    "    survey_trips['survey_name'] = survey_trips.user_input.apply(lambda sr: sr['trip_user_input']['data']['name'])\n",
+    "\n",
+    "    #gather the cols needed for charts and text\n",
+    "    survey_trips = survey_trips[['survey_name', 'user_id', 'user_input']]\n",
+    "    \n",
+    "    # transform data to meet a data format requirement\n",
+    "    all_composite_trips['sections'] = all_composite_trips['sections'].apply(lambda x: [x[0]['data']] if len(x) > 0 else x)\n",
+    "\n",
+    "    #only want trips with sections\n",
+    "    all_composite_trips = all_composite_trips[all_composite_trips['sections'].str.len() > 0]\n",
+    "    print(len(all_composite_trips), \"trips with sections\")\n",
+    "else:\n",
+    "    survey_trips = pd.DataFrame()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7b7001a9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#create translation dictionaries\n",
+    "opt_dict, quest_dict = build_dictionaries(sheet_list)\n",
+    "print(\"Questions dictionary:\\n\",quest_dict)\n",
+    "print(\"Options dictionary:\\n\", opt_dict)\n",
+    "\n",
+    "#color dictionary\n",
+    "color_map = scaffolding.mapping_color_surveys(opt_dict)\n",
+    "\n",
+    "#format survey trips into responses dataframe\n",
+    "df_responses = create_dataframe(survey_trips)\n",
+    "file_suffix = scaffolding.get_file_suffix(year, month, program)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93dbaa5e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#create the total dfs\n",
+    "total_dfs = {}\n",
+    "#for conditional surveys!\n",
+    "wrapped_config = {'survey_info': survey_info}\n",
+    "if 'buttons' in survey_info.keys():\n",
+    "    all_composite_trips['survey_name_prompted'] = all_composite_trips.apply(lambda row: conditional_surveys.survey_prompted_for_trip(row.to_dict(), wrapped_config), axis=1)\n",
+    "    \n",
+    "    for survey_name in list(sheet_list.keys()):\n",
+    "        if survey_name in all_composite_trips['survey_name_prompted'].unique():\n",
+    "            total_dfs[survey_name] = all_composite_trips[all_composite_trips['survey_name_prompted'] == survey_name]\n",
+    "        else:\n",
+    "            #never prompted\n",
+    "            total_dfs[survey_name] = pd.DataFrame()\n",
+    "else:\n",
+    "    survey_name = list(sheet_list.keys())[0] #there is only one if non-conditional\n",
+    "    total_dfs[survey_name] = all_composite_trips\n",
+    "    \n",
```

Comment:
can't we just use `pandas.groupby(...)` here?

Reply:
@shankari I'm not sure where you're proposing the `groupby`, can you explain a little more what you mean?